NPE for null links

Entry with the following format:

```json
"fields": {
  "link": null
}
```

Results in NPE being thrown:
```
java.lang.NullPointerException
	at com.contentful.java.cda.Util.extractNested(Util.java:86)
	at com.contentful.java.cda.ResourceUtils.getLinkId(ResourceUtils.java:155)
	at com.contentful.java.cda.ResourceUtils.resolveSingleLink(ResourceUtils.java:134)
	at com.contentful.java.cda.ResourceUtils.resolveLinks(ResourceUtils.java:63)
	at com.contentful.java.cda.ResourceFactory.sync(ResourceFactory.java:61)
	at com.contentful.java.cda.SyncQuery$1.call(SyncQuery.java:49)
	at com.contentful.java.cda.SyncQuery$1.call(SyncQuery.java:47)
	at rx.internal.operators.OperatorMap$1.onNext(OperatorMap.java:55)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:363)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:325)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onNext(OperatorMerge.java:800)
	at retrofit.RxSupport$2.run(RxSupport.java:56)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at retrofit.Platform$Base$2$1.run(Platform.java:94)
	at java.lang.Thread.run(Thread.java:745)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: retrofit.client.Response.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:104)
	at rx.internal.operators.OperatorMap$1.onNext(OperatorMap.java:58)
```

--- a/src/main/java/com/contentful/java/cda/ResourceUtils.java
+++ b/src/main/java/com/contentful/java/cda/ResourceUtils.java
@@ -145,6 +145,9 @@ final class ResourceUtils {
   }
 
   static String getLinkId(Object link) {
+    if (link == null) {
+      return null;
+    }
     if (link instanceof CDAResource) {
       // already resolved
       return ((CDAResource) link).id();

--- a/src/test/java/com/contentful/java/cda/LinkTest.java
+++ b/src/test/java/com/contentful/java/cda/LinkTest.java
@@ -50,4 +50,13 @@ public class LinkTest extends BaseTest {
 
     assertThat((List<String>) container.getField("symbols")).containsExactly("a", "b", "c");
   }
+
+  @Test @Enqueue(defaults = {
+      "links/space.json",
+      "links/content_types.json"
+  }, value = {
+      "links/sync_empty_links.json"
+  }) public void testEmptyLinks() throws Exception {
+    client.sync().fetch();
+  }
 }

--- a/src/test/resources/links/sync_empty_links.json
+++ b/src/test/resources/links/sync_empty_links.json
@@ -1,0 +1,47 @@
+{
+  "sys": {
+    "type": "Array"
+  },
+  "items": [
+    {
+      "sys": {
+        "space": {
+          "sys": {
+            "type": "Link",
+            "linkType": "Space",
+            "id": "75oea9kejr52"
+          }
+        },
+        "type": "Entry",
+        "contentType": {
+          "sys": {
+            "type": "Link",
+            "linkType": "ContentType",
+            "id": "2s9novBkP2G0oasUaG8kCs"
+          }
+        },
+        "id": "3vyEoAvlkk8yE4a8gCCkiu",
+        "revision": 1,
+        "createdAt": "2015-07-10T11:56:10.469Z",
+        "updatedAt": "2015-07-10T11:56:10.469Z"
+      },
+      "fields": {
+        "entry": {
+          "en-US": null
+        },
+        "asset": {
+          "en-US": null
+        },
+        "entries": {
+          "en-US": []
+        },
+        "assets": {
+          "en-US": []
+        },
+        "symbols": {
+          "en-US": []
+        }
+      }
+    }
+  ]
+}

